src/materials/ComputeFPIsolatorElasticity.C

```diff
@@ -0,0 +1,488 @@
+  Real vel2 = _basic_vel_old[_qp](2, 0) +
+              (_gamma / _beta) * (((_basic_def[_qp](2, 0) - _basic_def_old[_qp](2, 0)) / _dt) -
+                                  (_basic_vel_old[_qp](2, 0)));
+
+  // Calculate absolute velocity
+  Real velAbs =
+      sqrt(pow(vel1 / r_y * _basic_def[_qp](1, 0) + vel2 / r_z * _basic_def[_qp](2, 0), 2) +
+           pow(vel1, 2) + pow(vel2, 2));
+
+  // Check for uplift
+  Real is_uplift = 0.0;
+  Real kFactUplift = 1.0E-6;
+  if (_Fb[_qp](0, 0) >= 0.0)
+  {
+    _Fb[_qp](0, 0) = -0.0001 * _Fb[_qp](0, 0);
+    is_uplift = 1.0;
+  }
+
+  // Unit conversion for pressure to be used in the pressure factor computation
+  Real p_Unit_Convert; // To convert to MPa
+  if (_unit == 1)
+  {
+    p_Unit_Convert = 0.000001;
+  }
+  if (_unit == 2)
+  {
+    p_Unit_Convert = 0.001;
+  }
+  if (_unit == 3)
+  {
+    p_Unit_Convert = 1.0;
+  }
+  if (_unit == 4)
+  {
+    p_Unit_Convert = 1000.0;
+  }
+  if (_unit == 5)
+  {
+    p_Unit_Convert = 0.006894;
+  }
+  if (_unit == 6)
+  {
+    p_Unit_Convert = 6.894;
+  }
+  if (_unit == 7)
+  {
+    p_Unit_Convert = 0.00004788;
+  }
+  if (_unit == 8)
+  {
+    p_Unit_Convert = 0.04788;
+  }
+
+  // Calculate normal force
+  Real N = -_Fb[_qp](0, 0);
+
+  // Calculate instantaneous pressure
+  Real Inst_Pressure = fabs(N) / (_pi * _r_contact * _r_contact);
+
+  // Calculate displacement increment in the current step
+  _disp_currentstep = sqrt(pow((_basic_def[_qp](1, 0) - _basic_def_old[_qp](1, 0)), 2) +
+                           pow((_basic_def[_qp](2, 0) - _basic_def_old[_qp](2, 0)), 2));
+
+  // Calculate sliding velocity
+  _vel_currentstep = _disp_currentstep / (_vec_time[_t_step] - _vec_time[_t_step - 1]);
+
+  if (velAbs > 0.0)
+    _vel_currentstep = velAbs;
+
+  // Calculate pressure factor
+  if (_pressure_dependent)
+    _kpF = pow(0.7, ((Inst_Pressure - _p_ref) * p_Unit_Convert / 50.0));
+
+  // Calculate temperature factor
+  if (_temperature_dependent)
+  {
+    // Calculate flux
+    if (resultantU < (_r_contact * sqrt(_pi / 4)))
+      _vec_heatflux[_t_step] =
+          _mu_adj * (fabs(N) / (_pi * _r_contact * _r_contact)) * _vel_currentstep;
+
+    else
+      _vec_heatflux[_t_step] = 0.0;
+
+    _heatflux_center = _vec_heatflux[_t_step];
+
+    // Temperature at the sliding surface
+    Real temperature_change = 0.0;
+
+    // Evaluate the temperature integral
+    if (_t_step >= 1 && resultantU > 0.0)
+    {
+      Real dt_analysis = 0.000001;
+      Real tau = 0.000001;
+
+      for (int jTemp = 1; jTemp <= _t_step; jTemp++)
+      {
+        dt_analysis = _vec_time[_t_step] - _vec_time[_t_step - 1];
+        tau = _vec_time[jTemp];
+
+        temperature_change =
+            temperature_change +
+            sqrt(_diffusivity / _pi) * _vec_heatflux[_t_step - jTemp] * dt_analysis /
+                (_conductivity * sqrt(tau)); // Solving the integral for temperature increment
+      }
```

Comment:
This a convolution integral. So it is not possible to just add the temperature change to the previous step. The integral has to be solved at every timestep. Thought of many ways to avoid it, one of which is the use FFT based algorithms to solve convolution integral. But owing to complexity we resorted to solve it at every time step

Reply:
Thanks for the clarification